x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -427,12 +434,19 @@ export class TaskRunner<
         })
       );
     } catch (err) {
-      this.alertingEventLogger.setExecutionFailed(
-        `rule execution failure: ${ruleLabel}`,
-        err.message
-      );
+      if (
+        getReasonFromError(err) === RuleExecutionStatusErrorReasons.MaxAlerts ||
+        err.message.match(/Rule generated greater than \d alerts/)
+      ) {
```

Comment:
If the rule type executor doesn't do any error handling, `getReasonFromError` will return a reason that matches `RuleExecutionStatusErrorReasons.MaxAlerts`.

If the rule type executor catches the error and rethrows as `new Error(err)`, the reason will be stripped out but the error message remains so we can try to match that message.

Otherwise, if the rule type executor catches the error and throws a new error with its own error message, we are unable to differentiate that here and short-circuit the entire execution flow. Execution will end in an error rather than a warning.

Reply:
We could set a flag in the `alertFactory` to indicate that max alerts has been reached and check that here, making the assumption that if an error is thrown and the flag is `true`, the error came from trying to create too many alerts. Not sure how safe that assumption is.